src/main/java/se/ayoy/maven/plugins/licenseverifier/LicenseAbstractMojo.java

```diff
@@ -96,31 +98,52 @@
                 continue;
             }
 
-            AyoyArtifact licenseInfo = new AyoyArtifact(artifact);
+            toReturn.add(toAyoyArtifact(artifact, buildingRequest));
 
-            getLog().debug("Getting license for " + artifact.toString());
-            try {
-                buildingRequest.setProject(null);
+            Set<Artifact> transitiveArtifacts = resolveTransitiveArtifact(artifact);
 
-                MavenProject mavenProject = projectBuilder.build(artifact, buildingRequest).getProject();
-                if (mavenProject == null) {
-                    throw new NullPointerException("MavenProject retrieved from ProjectBuilder.build is null");
-                }
+            StringBuilder transitiveArtifactsList = new StringBuilder();
+            for (Artifact transitiveArtifact : transitiveArtifacts) {
+                toReturn.add(toAyoyArtifact(transitiveArtifact, buildingRequest));
+                transitiveArtifactsList.append(lineSeparator())
+                        .append(transitiveArtifact.toString());
+            }
+            if (getLog().isDebugEnabled() && !transitiveArtifacts.isEmpty()) {
+                getLog().debug("Verifying "
+                        + transitiveArtifacts.size()
+                        + " transitive artifacts for "
+                        + artifact.toString()
+                        + ":"
+                        + transitiveArtifactsList
+                );
+            }
+        }
 
-                List<License> licenses = mavenProject.getLicenses();
-                if (licenses == null) {
-                    throw new NullPointerException("Licenses is null, from mavenProject from " + artifact);
-                }
+        return toReturn;
+    }
 
-                licenseInfo.addLicenses(licenses);
+    private AyoyArtifact toAyoyArtifact(Artifact artifact, ProjectBuildingRequest buildingRequest) {
+        AyoyArtifact licenseInfo = new AyoyArtifact(artifact);
 
-                toReturn.add(licenseInfo);
-            } catch (ProjectBuildingException e) {
-                getLog().error(e.getMessage());
+        getLog().debug("Getting license for " + artifact.toString());
+        try {
+            buildingRequest.setProject(null);
+
+            MavenProject mavenProject = projectBuilder.build(artifact, buildingRequest).getProject();
+            if (mavenProject == null) {
+                throw new NullPointerException("MavenProject retrieved from ProjectBuilder.build is null");
             }
-        }
 
-        return toReturn;
+            List<License> licenses = mavenProject.getLicenses();
+            if (licenses == null) {
+                throw new NullPointerException("Licenses is null, from mavenProject from " + artifact);
+            }
+
+            licenseInfo.addLicenses(licenses);
+        } catch (ProjectBuildingException e) {
+            getLog().error(e.getMessage());
```

Comment:
@smuda 
Why we do not throw `MojoExecutionException`?
`ProjectBuildingException` is the internal error in Maven and we cannot do anything about it and thus ignoring it may succeed the build instead of expectantly to fail the build.

Reply:
If I remember correctly, the exception is thrown by projectBuilder.build. Exactly under what circumstances that fails I didn't know and wanted to be careful to do exactly what the plugin was supposed to do. 
It was not a strong opinion and I agree that a MojoExcecutionException should be thrown. I've added a commit with that change.